railties/lib/rails/env.rb

```diff
@@ -0,0 +1,19 @@
+module Rails
+  class << self
+    # Returns the current Rails environment.
+    #
+    #   Rails.env # => "development"
+    #   Rails.env.development? # => true
+    #   Rails.env.production? # => false
+    def env
+      @_env ||= ActiveSupport::StringInquirer.new(ENV["RAILS_ENV"] || ENV["RACK_ENV"] || "development")
+    end
+
+    # Sets the Rails environment.
+    #
+    #   Rails.env = "staging" # => "staging"
+    def env=(environment)
+      @_env = ActiveSupport::StringInquirer.new(environment)
```

Comment:
Perhaps move updating `RAILS_ENV` here as well, so rest of code can just do `Rails.env = new_env` without also assigning it to RAILS_ENV everywhere.